conformance/lang/expressions/logical-expressions/logical_and.balt

```diff
@@ -0,0 +1,732 @@
+    io:println(false && a == "text"? 2 : 1); // @error static type of each expression must be a subtype of boolean
+}
+
+Test-Case: error
+Description: Test static type of expression in logical expression via invalid types using conditional expression.
+Labels: logical-expr, conditional-expr
+
+function init() {
+    string a = "text";
+    boolean _ =  a == "text"? 2 : 1 && true; // @error static type of each expression must be a subtype of boolean
+    boolean _ =  a == "text"? 2 : 1 && false; // @error static type of each expression must be a subtype of boolean
+    boolean _ =  a == "text"? 2 : 1 && a == "text"? 2 : 1; // @error static type of each expression must be a subtype of boolean
+}
+
+Test-Case: error
+Description: Test static type of expression in logical expression via invalid types using conditional expression with paranthesis.
+Labels: logical-expr, conditional-expr
+
+function init() {
+    string a = "text";
+    boolean _ =  (a == "text"? 2 : 1) && true; // @error static type of each expression must be a subtype of boolean
+    boolean _ =  true && (a == "text"? 2 : 1); // @error static type of each expression must be a subtype of boolean
+    boolean _ =  false && (a == "text"? 2 : 1); // @error static type of each expression must be a subtype of boolean
+    boolean _ =  (a == "text"? 2 : 1) && false; // @error static type of each expression must be a subtype of boolean
+    boolean _ =  (a == "text"? 2 : 1) && (a == "text"? 2 : 1); // @error static type of each expression must be a subtype of boolean
+}
+
+Test-Case: error
+Description: Test static type of expression in logical expression via invalid types using checking expression.
+Labels: logical-expr, checking-expr
+
+function init() {
+    boolean _ =  true && check int:fromString("123"); // @error static type of each expression must be a subtype of boolean
+    boolean _ =  check int:fromString("123") && true; // @error static type of each expression must be a subtype of boolean
+    boolean _ =  check int:fromString("123") && check int:fromString("123"); // @error static type of each expression must be a subtype of boolean
+}
+
+Test-Case: error
+Description: Test static type of expression in logical expression via invalid types using trap expression.
+Labels: logical-expr, trap-expr
+
+function init() {
+    boolean _ =  true && trap divideNum(1, 0); // @error static type of each expression must be a subtype of boolean
+    boolean _ =  divideNum(1, 0) && true; // @error static type of each expression must be a subtype of boolean
+    boolean _ =  divideNum(1, 0) && divideNum(1, 0); // @error static type of each expression must be a subtype of boolean
+}
+
+function divideNum(int a, int b) returns int {
+    return a / b;
+}
+
+Test-Case: error
+Description: Test static type of expression in logical expression via invalid types using xml navigate expression.
+Labels: logical-expr, xml-navigate-expr
```

Comment:
Suggested labels are `xml-filter-expr` or `xml-step-expr`. Please update the usages of `xml-navigate-expr` to use the relevant specific label instead.